src/probnum/diffeq/odesolution.py

```diff
@@ -149,19 +154,22 @@ def __call__(self, t, smoothed=True):
         ----------
         t : float
             Location / time at which to evaluate the continuous ODE solution.
-        smoothed : bool, optional
-            If ``True`` (default) perform smooth interpolation. If ``False`` perform a
-            prediction from the previous location, without smoothing.
 
         Returns
         -------
         :obj:`RandomVariable`
             Probabilistic estimate of the continuous-time solution at time ``t``.
         """
-        out_rv = self._kalman_posterior(t, smoothed=smoothed)
-        out_rv = self._solver.undo_preconditioning(out_rv)
-        out_rv = self._proj_normal_rv(out_rv, 0)
-        return out_rv
+        out_rv = self._kalman_posterior(t)
+        if np.isscalar(t):
+            out_rv = self._solver.undo_preconditioning(out_rv)
+            return self._proj_normal_rv(out_rv, 0)
+        out_rvs = _RandomVariableList(
+            [self._solver.undo_preconditioning(out_rv_) for out_rv_ in out_rv]
+        )
+        return _RandomVariableList(
```

Comment:
As I understand it you're creating a RVList, just to iterate over it again and to do `self._proj_normal_rv`. Maybe it would be better to do this in one single loop?
That is
```python
        out_rvs = _RandomVariableList(
            [self._proj_normal_rv(self._solver.undo_preconditioning(out_rv_), 0) 
             for out_rv_ in out_rv]
        )
        return out_rvs
```
I guess "better" is the wrong word, the readability can be criticised (but I guess it would also look different through proper code formatting with Black), but it also saves an additional loop.

Even if you decide against it, there is no need for the first RVList and you could just have a python list.

Reply:
An alternative to this that comes to mind would be that `undo_preconditioning` and `_proj_normal_rv` could operate on both randomvariables and RVLists. Though I'm also not sure if that makes code cleaner since it makes things less explicit.

Reply:
You're right.